pkg/layer2/announcer.go

```diff
@@ -0,0 +1,136 @@
+package layer2
+
+import (
+	"github.com/go-logr/logr"
+	portererror "github.com/kubesphere/porter/pkg/errors"
+	"github.com/kubesphere/porter/pkg/kubeutil"
+	"net"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"time"
+)
+
+// Announce is used to "announce" new IPs mapped to the node's MAC address.
```

Comment:
Please add a comment about `Referring to the code in https://github.com/metallb/metallb/tree/main/internal/layer2`

Reply:
ok,  but there are many differences

Reply:
OK, maybe this is better
```go
// Referenced part of the code in https://github.com/metallb/metallb/tree/main/internal/layer2
```